### PR TITLE
Add check for large for loops

### PR DESCRIPTION
**Pull request by BoyaWu10**
By offline discussion, we are going to limit the number of the list in the for loop by the `maxIterations` parameter provided by dotliquid. Currently the value is set to 100000.

### CODE REVIEW

**Review by sowu880 — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ namespace Microsoft.Health.Fhir.Liquid.Converter.Processors
     public abstract class BaseProcessor : IFhirConverter
     {
         private readonly ProcessorSettings _settings;
+        private const int _maxIterations = 100000;
```
Comment (qiwjin) [inline src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs:21]: can help doc this limitation?

**Review by qiwjin — APPROVED** [review]

**Comment by BoyaWu10** [comment]
Hi @ginalee-dotcom, could you please help create a PR to update the document about this change? The maximum iteration of for-loop in liquid templates is now set to 100,000. Previously there's no limitation about this. If the limitation is reached, an exception will be thrown with detailed message. This is to help make sure our engine returns a result in time.

File: src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs (hunk 1 above)
Comment (BoyaWu10) [reply src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs:21]: Hi @ginalee-dotcom, could you please help create a PR to update the document about this change? The maximum iteration of for-loop in liquid templates is now set to 100,000. Previously there's no limitation about this. If the limitation is reached, an exception will be thrown with detailed message. This is to help make sure our engine returns a result in time.

**Review by moria97 — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs [hunk 2]
```diff
@@ -43,7 +44,7 @@ protected virtual Context CreateContext(ITemplateProvider templateProvider, IDic
                 outerScope: new Hash(),
                 registers: Hash.FromDictionary(new Dictionary<string, object> { { "file_system", templateProvider.GetTemplateFileSystem() } }),
                 errorsOutputMode: ErrorsOutputMode.Rethrow,
-                maxIterations: 0,
+                maxIterations: _maxIterations,
```
Comment (ACMoretxj) [inline src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs:47]: Will this configuration effect for the following template?

```liquid
{
    {% for i in (1..10000) -%}
        {% for j in (1..10000) -%}
            "value": [{% for k in (1..10000) -%}{{ k }},{% endfor -%}]
        {% endfor -%}
    {% endfor -%}
}
```

**Review by yankunhuang-pku — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs (hunk 2 above)
Comment (BoyaWu10) [reply src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs:47]: No, the limitation only works for single for-loop. An unit test is added to show this.

Comment (BoyaWu10) [reply src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs:47]: Besides, if the nested for-loops take too much time, it should be killed by timeout setting, which is set by FHIR Server.

**Review by ACMoretxj — APPROVED** [review]

**Pull request merged**